### DISCUSSION TITLE
[Bug]: 'HTMLCanvasElement': Tainted canvases may not be exported

### DISCUSSION THREAD

**Discussion by zmiju** [fabricjs/fabric.js#8669] [Q&A] [answered]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.3.0

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://stackblitz.com/edit/angular-pggk9r?file=src%2Fmain.ts

### Steps To Reproduce

1. Click "change"
2. Click "download"
3. Bug occurs

### Expected Behavior

I want to download canvas with custom image on it.

### Actual Behavior

When I try to download canvas with image, bug occurs: 'HTMLCanvasElement': Tainted canvases may not be exported

### Error Message & Stack Trace

```bash
'HTMLCanvasElement': Tainted canvases may not be exported
```

**Comment by ShaMan123** [comment, marked as answer]
cross origin seems to be, pass `crossOrigin` as options to `loadImage`

**Reply by ShaMan123** [reply]
not related to fabric

**Reply by zmiju** [reply]
Thanks for comment, but how to fix it?

**Comment by zmiju** [comment]
I have found solution. Maybe someone find it useful.

```
    let c = document.createElement('img');
    let i = new fabric.Image(c);
    c.onload = () => {
      i = new fabric.Image(c);
      this.canvas.add(i);
    };
    c.setAttribute('crossOrigin', 'anonymous');
    c.src = "<image url>";
```

**Reply by ShaMan123** [reply]
~~This is great!
I believe it is a bug fix.
Would you PR?
I can point out where and what to do.~~

Was completely confused when reading the code

**Reply by zmiju** [reply]
Could you explain what is PR? :)

**Reply by ShaMan123** [reply]
Right
PR stands for Pull Request, read about in github docs.
It is a change proposal to the repository, [here](https://github.com/fabricjs/fabric.js/pulls) are fabric's
It is a great way to track and discuss changes, a coding convension.
I am assuming you are not familiar with git. You should. I would say must.
It is the difference between walking in circles or seeing clearly the path forward since you have the code diff to guide you.
I would start by installing github desktop and start working with that.

**Reply by zmiju** [reply]
Thanks for clarification.
I only use github as a repository of my own software, so I am rather a beginner.

**Comment by ShaMan123** [comment]
updated thread
